lib/parallel_tests/test/runner.rb

```diff
@@ -209,16 +209,26 @@ def sort_by_filesize(tests)
         end
 
         def find_tests(tests, options = {})
+          tests_list = all_tests(tests, options)
+          exclude_tests(tests_list, options)
+        end
+
+        def all_tests(tests, options)
           (tests || []).map do |file_or_folder|
             if File.directory?(file_or_folder)
               files = files_in_folder(file_or_folder, options)
-              files.grep(options[:suffix]||test_suffix).grep(options[:pattern]||//)
+              files.grep(options[:suffix] || test_suffix)
+                .grep(options[:pattern] || //)
             else
               file_or_folder
             end
           end.flatten.uniq
         end
 
+        def exclude_tests(tests_list, options = {})
+          tests_list.reject { |file| file =~ (options[:exclude_pattern] || //) }
```

Comment:
- `//` matches everything
 - use either reject! or grep since they are more efficient
 - do as little work as possible in the inner loop (use local variable and not option lookup)

```suggestion
          if regex = options[:exclude_pattern]
            tests_list -= tests_list.grep(regex)
          end
```

Reply:
Addressed. Thanks for the quick feedback!